### PR TITLE
Applicative

### PR DESCRIPTION
**Pull request by gabriel128**

### CODE REVIEW

File: src/Course/Monad.hs [hunk 1]
```diff
@@ -32,48 +29,39 @@ infixr 1 =<<
...
 -- [1,1,2,2,3,3]
 instance Monad List where
-  (=<<) ::
-    (a -> List b)
-    -> List a
-    -> List b
-  (=<<) =
-    error "todo: Course.Monad (=<<)#instance List"
+  (=<<) :: (a -> List b) -> List a -> List b
+  f =<< xs = flatten . map f $ xs
```
Comment (JulianLeviston) [inline src/Course/Monad.hs:46]: Nice! we also have `concatMap`, so `f =<< xs = concatMap f xs` which... we can rewrite as
`(=<<) = concatMap` ;-)

File: src/Course/Monad.hs [hunk 2]
```diff
@@ -32,48 +29,39 @@ infixr 1 =<<
...
 instance Monad ((->) t) where
-  (=<<) ::
-    (a -> ((->) t b))
-    -> ((->) t a)
-    -> ((->) t b)
-  (=<<) =
-    error "todo: Course.Monad (=<<)#instance ((->) t)"
+  -- (=<<) :: (a -> (t -> b)) -> (t -> a) -> (t -> b)
+  (=<<) :: (a -> ((->) t b)) -> ((->) t a) -> ((->) t b)
+  (f =<< g) x = f (g x) x
```
Comment (JulianLeviston) [inline src/Course/Monad.hs:64]: nice! this took me ages to realise how to do this the first time someone explained it to me... :)

File: src/Course/Monad.hs (hunk 1 above)
Comment (gabriel128) [reply src/Course/Monad.hs:46]: we have that one? that was going to be my first approach but I thought it wasn't defined

File: src/Course/Monad.hs (hunk 2 above)
Comment (gabriel128) [reply src/Course/Monad.hs:64]: haha, to be fair I wrote a few proofs of the reader monad in the past, I remember it took me some time to understand it too

File: src/Course/Monad.hs (hunk 1 above)
Comment (JulianLeviston) [reply src/Course/Monad.hs:46]: Ah it might be called `flatMap`... but yeah I think so... it's in `List`, IIRC. But either way, it's good 👍

**Pull request merged**